CONTRIBUTING.md

```diff
@@ -118,23 +118,48 @@ This section applies to both Python versions 2 and 3.
 
 It is recommended to use `virtualenv` to develop in a sandbox environment:
 
-```
+```sh
 virtualenv venv
 . venv/bin/activate
-pip install -r tests/requirement_tests.txt
+
+# Install neurodamus in development mode
+pip install -e .
+
+# Install test requirements
+pip install -r tests/requirements.txt
 ```
 
-## Build
+## Testing
 
-Run the following command to build incrementally the project: `pip install -e .`
+There are several test groups in Neurodamus, from plain unit tests to integration and system tests.
 
-## Test
+While developing you may want to run unit tests very frequently and thus we suggest running the base
+tests using pytest directly from the dev environment.
+```sh
+pytest tests/unit
+```
+
+For the next stage testing we suggest using the provided tox environments
 
-Run the following command to run the Python unit-tests: `pytest tests`
+```sh
+# Integration tests
+tox -e integration
+```
```

Comment:
Can we make it clear what's the criteria of `integration` tests w.r.t others ? I see we still include `common` mod files in `build_ndcore.sh`.  If we move `test_plugin.py` to the integration-e2e, maybe we can run `integration` tests  without additional mod files?

Reply:
The lines can be a bit blurry because we are trying to fit existing tests without massive changes (and still this is already getting large).
Having said that, even if we try to stay consistent with testing guidelines, where integration tests should test a part of the system, and for that we might require BlueBrain synapses, but it's not rigid. So I don't oppose moving `test_plugin.py` to integration-e2e.

For other groups, I think integration-e2e is for whole-system tests (hence we need neocortex), whereas scientific tests is the previous group whcih tests specifically features the scientists care for.
Anyway, open for feedback.

Reply:
The `test_connection.py` file ? It is in "integration_e2e".  Is it because we need to load cells?
I agree it doesn't hurt to build a special with synapses for `integration` tests. Just wondering if we will use it.